bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/internal/ThingManagerImpl.java

```diff
@@ -1331,31 +1049,200 @@ private void updateChildThingStatusForDisabledBridges(Bridge bridge) {
     private void persistThingEnableStatus(ThingUID thingUID, boolean enabled) {
         logger.debug("Thing with UID {} will be persisted as {}.", thingUID, enabled ? "enabled." : "disabled.");
         if (enabled) {
-            // Clear the disabled thing storage. Otherwise the handler will NOT be initialized later.
-            storage.remove(thingUID.getAsString());
+            // Clear the disabled thing storage. Otherwise, the handler will NOT be initialized later.
+            disabledStorage.remove(thingUID.getAsString());
         } else {
             // Mark the thing as disabled in the storage.
-            storage.put(thingUID.getAsString(), "");
+            disabledStorage.put(thingUID.getAsString(), "");
         }
     }
 
     @Override
     public boolean isEnabled(ThingUID thingUID) {
-        Thing thing = this.things.get(thingUID);
+        Thing thing = things.get(thingUID);
         if (thing != null) {
             return thing.isEnabled();
         }
 
         logger.debug("Thing with UID {} is unknown. Will try to get the enabled status from the persistent storage.",
                 thingUID);
-        return !isDisabledByStorage(thingUID);
+        return !disabledStorage.containsKey(thingUID.getAsString());
     }
 
-    private boolean isDisabledByStorage(ThingUID thingUID) {
-        return storage.containsKey(thingUID.getAsString());
+    private boolean checkAndPerformUpdate(Thing thing, ThingHandlerFactory factory) {
+        final int currentThingTypeVersion = Integer
```

Comment:
You will need a different treatment for textually defined things. With the current implementation, they are always considered to be version "0", which means that the logic below is always performed (and the info logging happens on every start again and again).
Instead, for textual things we can assume that they are always of the latest thing-type version, since their channels are typically created on startup based on the latest thing-type definition.

Reply:
Luckily this is not the case. Since textually defined things are re-created each time the `.things` file is read, they are automatically created with the latest version (given the developer added the correct version property to the thing-type definition).

Reply:
> given the developer added the correct version property to the thing-type definition

Why is this necessary at all? Shouldn't we simply assume that the thing-type is always the latest version, which can be derived from the update information? I do not really see a need to additionally specify the version in the thing-type xml - this is rather something that might easily be forgotten or getting out of sync with the update files.

Reply:
I added a check and only update if the provider is a `ManagedProvider`